github/Team.py

```diff
@@ -234,13 +235,20 @@ def get_repo_permission(self, repo):
         except UnknownObjectException:
             return None
 
+    @deprecated(
+        reason="""
+        Team.set_repo_permission() is deprecated, use
+        Team.update_team_repository() instead.
```

Comment:
I wonder if this fits on one line. It's no issue if it does not.

Reply:
Done in https://github.com/PyGithub/PyGithub/pull/1509/commits/076e25776ce84f4dc96f7be9aba5ee3e120527af